### ISSUE TITLE
NullReferenceException not thrown for nullref in RhpInterfaceDispatch2

### ISSUE THREAD

**Issue by MichalStrehovsky** [dotnet/runtimelab#221]
Loader/classloader/methodoverriding/regressions/549411/exploit

We tear down the process instead.

**Comment by MichalStrehovsky** [comment]
.NET Native does a null check before the interface call:

```
00007FFE35DFC0B4 48 8D 05 C5 FE F9 FF lea         rax,[$0_App1::MainPage.foo (07FFE35D9BF80h)]  
00007FFE35DFC0BB 48 8B 08             mov         rcx,qword ptr [rax]  
00007FFE35DFC0BE 80 39 00             cmp         byte ptr [rcx],0  
00007FFE35DFC0C1 4C 8D 15 B8 FB FB FF lea         r10,[InterfaceDispatchCell000ABC80_$0_IFoo_slot_0 (07FFE35DBBC80h)]  
00007FFE35DFC0C8 41 FF 12             call        qword ptr [r10] 
```

I guess we should do the same.

**Comment by jkotas** [comment]
Related change: https://github.com/dotnet/runtime/pull/35058/

> I guess we should do the same.

That. Or convert NullRefs in the dispatch helper to NullReferenceExceptions like what CoreCLR does.

**Comment by MichalStrehovsky** [comment]
Fixed in #239.
